modules/localgov_microsites_group_term_ui/src/Plugin/EntityReferenceSelection/GroupTermSelection.php

```diff
@@ -0,0 +1,114 @@
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   *   The module handler service.
+   * @param \Drupal\Core\Session\AccountInterface $current_user
+   *   The current user.
+   * @param \Drupal\Core\Entity\EntityFieldManagerInterface $entity_field_manager
+   *   The entity field manager.
+   * @param \Drupal\Core\Entity\EntityTypeBundleInfoInterface $entity_type_bundle_info
+   *   The entity type bundle info service.
+   * @param \Drupal\Core\Entity\EntityRepositoryInterface $entity_repository
+   *   The entity repository.
+   * @param \Drupal\domain_group\DomainGroupResolverInterface $domain_group_resolver
+   *   The domain group resolver service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, ModuleHandlerInterface $module_handler, AccountInterface $current_user, EntityFieldManagerInterface $entity_field_manager, EntityTypeBundleInfoInterface $entity_type_bundle_info = NULL, EntityRepositoryInterface $entity_repository, DomainGroupResolverInterface $domain_group_resolver) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_type_manager, $module_handler, $current_user, $entity_field_manager, $entity_type_bundle_info, $entity_repository);
+
+    $this->domainGroupResolver = $domain_group_resolver;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('module_handler'),
+      $container->get('current_user'),
+      $container->get('entity_field.manager'),
+      $container->get('entity_type.bundle.info'),
+      $container->get('entity.repository'),
+      $container->get('domain_group_resolver')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getReferenceableEntities($match = NULL, $match_operator = 'CONTAINS', $limit = 0) {
+    $options = parent::getReferenceableEntities($match, $match_operator, $limit);
+
+    // Try and load the group.
+    $group_id = \Drupal::service('domain_group_resolver')->getActiveDomainGroupId();
+    if ($group_id) {
+      $group = \Drupal::entityTypeManager()->getStorage('group')->load($group_id);
+    }
+    if (empty($group)) {
+      $group = \Drupal::request()->attributes->get('group');
+    }
+
+    // Only show group terms if there's a group.
+    if ($group instanceof GroupInterface) {
```

Comment:
Guess rewriting and changing the access check of the parent wasn't an/the best option?
Bit of an edge issue but I think the parent deals with removing the hierarchy if the term above isn't visible. Not sure what's going to happen here. Maybe we just make a follow-up.

Reply:
Just adding an access check didn't work. Not sure why, but the check always came back TRUE. Once we started loading the group it seemed easier to call the parent and then filter that.

Good call about the hierarchy. The UI for managing group terms doesn't actually deal with the hierarchy as it's a flat view. although it is possible to add a parent term on the term edit form.

Reply:
Follow-up https://github.com/localgovdrupal/localgov_microsites_group/issues/176